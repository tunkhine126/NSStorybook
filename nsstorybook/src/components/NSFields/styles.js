import { makeStyles } from '@material-ui/core/styles';
import colors from '../../global-styles';
import styled from 'styled-components';

export const styles = makeStyles(() => ({
  caption: {
    fontSize: '1rem',
    fontWeight: 400,
    marginTop: 10,
    color: colors.TEXT.light,
  },
  dropzone: {
    marginTop: 30,
    padding: 25,
    border: `2px dashed ${colors.TEXT.medium}`,
    borderRadius: 2,
    '&:hover': {
      borderColor: colors.STATE.active,
    },
    outline: 'none',
    cursor: 'pointer',
    maxWidth: 300,
  },
  icon: {
    fontSize: 40,
    color: colors.TEXT.light,
  },
  img: {
    width: '100%',
  },
  overlay: {
    transition: '.5s ease',
    opacity: 0,
    position: 'absolute',
    top: '50%',
    left: ' 50%',
    transform: 'translate(-50%, -50%)',
    textAlign: 'center',
  },
  imgContainer: {
    cursor: 'auto',
    position: 'relative',
    '&:hover img': {
      opacity: 0.3,
    },
    '&:hover .img-overlay': {
      opacity: 1,
    },
  },
  deleteBtn: {
    '&:hover': {
      color: colors.STATUS.negative,
    },
  },
  deleteText: {
    fontWeight: 600,
    color: colors.TEXT.normal,
  },
}));

export const SelectWrapper = styled.div`
  margin-bottom: 8px;
  .MuiFormLabel-root {
    font-weight: normal;
  }
  .MuiInputBase-root {
    height: auto;
  }
  .MuiFormControl-root {
    margin-top: 16px;
    margin-bottom: 8px;
  }
`;
