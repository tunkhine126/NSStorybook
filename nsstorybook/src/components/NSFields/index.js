/**
 *
 * NSFields
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';

import DatePickerValidation from 'components/shared/DatePickerValidation';
import TimePickerValidation from 'components/shared/TimePickerValidation';
import SelectValidation from 'components/shared/SelectValidation';
import MultiSelectValidation from 'components/shared/MultiSelectValidation';
import InputValidation from 'components/shared/InputValidation';
import Photo from './photo';

import { SelectWrapper } from './styles';

function NSFields({
  field,
  fieldType,
  label,
  initialValue,
  showPrefix = false,
  icon,
  position,
  values = {},
  options = [],
  onchange = () => {},
  ...rest
}) {
  let preFixAttr = {};

  if (showPrefix) {
    preFixAttr = {
      position,
      icon,
    };
  }

  switch (fieldType) {
    case 'short_text':
      return (
        <InputValidation
          field={field}
          label={label}
          initialValue={initialValue}
          type="text"
          maxWidth="none"
          {...preFixAttr}
          {...rest}
        />
      );
    case 'paragraph_text':
      return (
        <InputValidation
          field={field}
          label={label}
          initialValue={initialValue}
          multiline
          type="text"
          maxWidth="none"
          rows="3"
          {...rest}
        />
      );
    case 'number':
      return (
        <InputValidation
          field={field}
          label={label}
          initialValue={initialValue}
          {...preFixAttr}
          type="number"
          maxWidth="none"
          {...rest}
        />
      );
    case 'date':
      return (
        <DatePickerValidation
          field={field}
          label={label}
          initialValue={initialValue}
          fullWidth
          clearable
          disableFuture
          customFormat="MM-dd-yyyy"
          placeholder={`${moment().format('MM-DD-YYYY')}`}
          variant="dialog"
          inputVariant="filled"
          margin="normal"
          {...rest}
        />
      );
    case 'time':
      return (
        <TimePickerValidation
          field={field}
          label={label}
          initialValue={initialValue}
          fullWidth
          clearable
          placeholder="08:00 AM"
          inputVariant="filled"
          margin="normal"
          {...rest}
        />
      );
    case 'single_select':
      return (
        <SelectWrapper>
          <SelectValidation
            field={field}
            label={label}
            initialValue={initialValue}
            values={values}
            items={options}
            onChange={onchange}
            type="select"
            maxWidth="none"
            padding="25px 18px 10px 13px"
            {...rest}
          />
        </SelectWrapper>
      );
    case 'multi_select':
      return (
        <SelectWrapper>
          <MultiSelectValidation
            field={field}
            label={label}
            initialValue={initialValue}
            values={values}
            onChange={onchange}
            items={options}
            type="select"
            maxWidth="none"
            padding="25px 18px 10px 13px"
            {...rest}
          />
        </SelectWrapper>
      );
    case 'likert':
      return (
        <SelectWrapper>
          <SelectValidation
            field={field}
            label={label}
            initialValue={initialValue}
            values={values}
            items={options}
            onChange={onchange}
            type="select"
            maxWidth="none"
            padding="25px 18px 10px 13px"
            {...rest}
          />
        </SelectWrapper>
      );
    case 'photo':
      return (
        <Photo
          field={field}
          initialValue={initialValue}
          label={label}
          {...rest}
        />
      );
    case 'audio':
      return 'audio';
    case 'video':
      return 'video';
    case 'coordinates':
      return (
        <InputValidation
          field={field}
          label={label}
          initialValue={initialValue}
          maxWidth="none"
          type="text"
          {...rest}
        />
      );
    default:
      return null;
  }
}

NSFields.propTypes = {
  field: PropTypes.string.isRequired,
  fieldType: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  initialValue: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.object,
    PropTypes.array,
  ]),
  position: PropTypes.string,
  showPrefix: PropTypes.bool,
  icon: PropTypes.oneOfType([PropTypes.element, PropTypes.func]),
  values: PropTypes.object,
  options: PropTypes.array,
  onchange: PropTypes.func,
};

export default NSFields;
