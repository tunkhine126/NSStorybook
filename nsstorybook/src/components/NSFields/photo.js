/**
 *
 * Photo Field Type
 *
 */

import React, { useCallback } from 'react';
import { asField } from 'informed';
import clsx from 'clsx';
import { injectIntl, intlShape } from 'react-intl';
import { useDropzone } from 'react-dropzone';

import Typography from '@material-ui/core/Typography';
import PhotoCameraIcon from '@material-ui/icons/PhotoCamera';
import Grid from '@material-ui/core/Grid';
import IconButton from '@material-ui/core/IconButton';
import DeleteIcon from '@material-ui/icons/Delete';

import { generateImg } from 'utils/helpers';
import { inputStyles } from 'global-styles';
import globalMessage from 'messages';
import { styles } from './styles';

const Photo = asField(({ fieldState, fieldApi, ...props }) => {
  const { value } = fieldState;
  const { setValue, setTouched } = fieldApi;
  const { label, intl, ...rest } = props;
  const classes = styles();
  const globalClasses = inputStyles(fieldState.error)();

  const getBase64 = (file, cb) => {
    const reader = new FileReader();
    reader.readAsDataURL(file);
    reader.onload = () => cb(reader.result);
    reader.onerror = error => {
      console.log('Error converting file to base64!: ', error);
    };
  };

  const onDrop = useCallback(
    acceptedFile => {
      getBase64(acceptedFile[0], result => {
        setValue(generateImg(result));
        setTouched(true);
      });
    },
    [setTouched, setValue]
  );

  const { getRootProps, getInputProps, isDragActive } = useDropzone({
    onDrop,
    accept: 'image/jpeg, image/png, image/tiff, image/ief, image/gif',
  });

  // These two props are not native and are not needed
  delete rest.forwardedRef;
  delete rest.debug;

  return (
    <>
      <div {...getRootProps({ className: clsx(classes.dropzone, 'dropzone') })}>
        <input {...getInputProps()} />
        <Grid
          container
          direction="column"
          justify="flex-start"
          alignItems="center"
        >
          {value ? (
            <Grid
              item
              className={classes.imgContainer}
              onClick={e => {
                e.stopPropagation();
              }}
            >
              <img
                src={value}
                alt={label}
                className={clsx(classes.img, 'uploaded-image')}
              />

              <Grid
                container
                direction="column"
                justify="flex-start"
                alignItems="center"
                className={clsx(classes.overlay, 'img-overlay')}
              >
                <Typography gutterBottom className={classes.deleteText}>
                  {intl.formatMessage(globalMessage.removeImg)}
                </Typography>
                <IconButton
                  className={classes.deleteBtn}
                  aria-label="delete photo"
                  onClick={() => {
                    setValue('');
                  }}
                >
                  <DeleteIcon />
                </IconButton>
              </Grid>
            </Grid>
          ) : (
            <>
              <PhotoCameraIcon className={classes.icon} />
              <Typography
                gutterBottom
                variant="body1"
                className={classes.caption}
              >
                {isDragActive
                  ? intl.formatMessage(globalMessage.dropFile)
                  : label}
              </Typography>
            </>
          )}
        </Grid>
      </div>
      {fieldState.error ? (
        <small className={clsx(globalClasses.error, globalClasses.marginTop)}>
          {fieldState.error}
        </small>
      ) : null}
    </>
  );
});

Photo.propTypes = {
  intl: intlShape.isRequired,
};

const intlPhoto = injectIntl(Photo);

export default intlPhoto;
