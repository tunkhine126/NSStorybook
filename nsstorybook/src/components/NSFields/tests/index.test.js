/**
 *
 * Tests for NSFields
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from '@testing-library/react';
// import { act } from 'react-dom/test-utils';
// import 'jest-dom/extend-expect'; // add some helpful assertions

import TestingComponent from 'components/shared/Testing';
import NSFields from '../index';

describe('<NSFields />', () => {
  const Prefix = () => <p style={{ minWidth: 50 }}>TEST-FAMILY-</p>;
  const singleValues = { item: 'red' };
  const multiValues = { item: ['red'] };
  const items = [{ value: 'blue' }, { value: 'red' }, { value: 'green' }];

  it('Expect no errors rending NSButton', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields
          fieldType="short_text"
          field="shortText"
          label="Short Text"
          position="start"
          showPrefix
          icon={Prefix}
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Paragraph Text Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields
          fieldType="paragraph_text"
          field="paragrapTest"
          label="Paragraph Text"
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Number Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields fieldType="number" field="number" label="Number Field" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Date Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields fieldType="date" field="date" label="Date Field" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Time Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields fieldType="time" field="time" label="Time Field" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Single Select Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields
          fieldType="single_select"
          field="singleSelect"
          label="Single Select Field"
          values={singleValues}
          options={items}
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Likert Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields
          fieldType="likert"
          field="likert"
          label="Likert Field"
          values={singleValues}
          options={items}
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Multi Select Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields
          fieldType="multi_select"
          field="mulitSelect"
          label="Multi Select Field"
          values={multiValues}
          options={items}
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Photo Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields
          fieldType="photo"
          field="photo"
          initialValue="iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8z8BQDwAEhQGAhKmMIQAAAABJRU5ErkJggg=="
          label="Photo Field"
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Audio Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields fieldType="audio" field="audio" label="Audio Field" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Video Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields fieldType="video" field="video" label="Video Field" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render Coordinates Field', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields
          fieldType="coordinates"
          field="coordinates"
          label="Coordinates Field"
        />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  it('Should render null', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(
      <TestingComponent>
        <NSFields fieldType="null" field="null" label="null" />
      </TestingComponent>
    );
    expect(spy).not.toHaveBeenCalled();
  });

  // it('invoke onDrop when drop event occurs', () => {
  //   const spy = jest.spyOn(global.console, 'error');
  //   const ui = (
  //     <TestingComponent>
  //       <NSFields fieldType="photo" field="photo" label="Photo Field" />
  //     </TestingComponent>
  //   );
  //   const { container } = render(ui);
  //   const file = createFile('dogs.jpg', 500, 'image/jpeg');

  //   const data = mockData([file]);
  //   const dropzone = container.querySelector('.dropzone');

  //   function dispatchEvt(node, type, d) {
  //     const event = new Event(type, { bubbles: true });
  //     Object.assign(event, d);
  //     fireEvent(node, event);
  //   }

  //   function mockData(files) {
  //     return {
  //       dataTransfer: {
  //         files,
  //         items: files.map(f => ({
  //           kind: 'file',
  //           type: f.type,
  //           getAsFile: () => f,
  //         })),
  //         types: ['Files'],
  //       },
  //     };
  //   }

  //   function createFile(name, size, type) {
  //     const f = new File([], name, { type });
  //     Object.defineProperty(f, 'size', {
  //       get() {
  //         return size;
  //       },
  //     });
  //     return f;
  //   }

  //   act(() => {
  //     dispatchEvt(dropzone, 'drop', data);
  //   });

  //   expect(spy).not.toHaveBeenCalled();
  // });

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <TestingComponent>
        <NSFields
          fieldType="short_text"
          field="shortText"
          label="Short Text"
          position="start"
          showPrefix
          icon={Prefix}
        />
      </TestingComponent>
    );
    expect(firstChild).toMatchSnapshot();
  });
});
