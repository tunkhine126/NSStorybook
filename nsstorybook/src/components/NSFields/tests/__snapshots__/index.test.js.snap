// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NSFields /> Should render and match the snapshot 1`] = `
<div
  class="styles__ComponentStyle-zscvjs-0 iXrycp"
>
  <div
    class="MuiFormControl-root MuiTextField-root makeStyles-input-973 makeStyles-root-977 MuiFormControl-marginNormal"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled"
      data-shrink="true"
      for="shortText"
      id="shortText-label"
    >
      Short Text
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root makeStyles-root-978 MuiInputBase-formControl MuiInputBase-adornedStart MuiFilledInput-adornedStart"
    >
      <div
        class="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionStart"
      >
        <p
          style="min-width: 50px;"
        >
          TEST-FAMILY-
        </p>
      </div>
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedStart MuiFilledInput-inputAdornedStart"
        id="shortText"
        name="shortText"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
